### PR TITLE
API Server: Override k8s api server profile contention flag with custom ones

### PR DESCRIPTION
**Pull request by marefr**
Following changes from https://github.com/grafana/grafana/pull/90048. This adds similar changes to the api server to allow block and mutex profiling rates to be configured.

Updates SDK to v0.238.0

### CODE REVIEW

File: pkg/services/apiserver/standalone/options/profiling.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	}
+}
+
+func (o *ProfilingOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&o.blockProfilingRate, "grafana.profiling.block-rate", 0, "Controls the fraction of goroutine blocking events that are reported in the blocking profile. The profiler aims to sample an average of one blocking event per rate nanoseconds spent blocked. To turn off profiling entirely, use 0.")
+	fs.IntVar(&o.mutexProfilingRate, "grafana.profiling.mutex-rate", 0, "Controls the fraction of mutex contention events that are reported in the mutex profile. On average 1/rate events are reported. To turn off mutex profiling entirely, use 0.")
+}
+
+func (o *ProfilingOptions) Validate() []error {
+	return nil
```
Comment (JohnnyQQQQ) [inline pkg/services/apiserver/standalone/options/profiling.go:29]: Suggestion: should we validate that both rates are not negative?

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM 🚀  one suggestion

File: pkg/services/apiserver/standalone/options/profiling.go (hunk 1 above)
Comment (marefr) [reply pkg/services/apiserver/standalone/options/profiling.go:29]: We didn't do that in https://github.com/grafana/grafana/pull/90048. In my head a negative value shouldn't have any impact besides runtime treating it as 0?

Comment (JohnnyQQQQ) [reply pkg/services/apiserver/standalone/options/profiling.go:29]: i might come handy if someone by accident misconfigures it, but not a blocker, feel free to merge

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**